app/javascript/packs/document-capture-welcome.ts

```diff
@@ -0,0 +1,67 @@
+import { hasCamera, isCameraCapableMobile } from '@18f/identity-device';
+
+const GRACE_TIME_FOR_CAMERA_CHECK_MS = 2000;
+
+function delay(ms: number): Promise<void> {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+function addFormInputsForMobileDeviceCapabilities() {
+  const form: HTMLFormElement | null = document.querySelector('.js-consent-continue-form');
+
+  if (!form) {
+    return;
+  }
+
+  if (!isCameraCapableMobile()) {
+    return;
+  }
+
+  // The check for a camera on the device is async -- kick it off here and intercept
+  // submit() to ensure that it completes in time.
+  let cameraPresent: boolean | undefined;
+  const cameraCheckPromise = hasCamera().then((result: boolean) => {
+    cameraPresent = !!result;
+
+    if (!cameraPresent) {
+      // Signal to the backend that this is a mobile device, but no camera is present
+      const ncInput = document.createElement('input');
+      ncInput.type = 'hidden';
+      ncInput.name = 'no_camera';
+      form.appendChild(ncInput);
+    }
```

Comment:
Some further context on this: On the backend, if we detect `params[:no_camera]`, we will redirect the user to an error screen, since they're on a mobile device with no camera and have no way to take pictures of their ID.